### PR TITLE
feat(atomic): atomic insight user actions toggle component created

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5617](https://coveord.atlassian.net/browse/SFINT-5617)

-  The `atomic-insight-user-actions-toggle` component displays a button that opens a modal containing the user actions timeline component.

- The `atomic-insight-user-action-toggle` component uses the `atomic-insight-history-toggle` component to display the icon button for user actions.

- The `atomic-insight-user-actions-modal` is automatically created as a child of the `atomic-insight-interface` when the `atomic-insight-user-actions-toggle` is initialized.

https://github.com/user-attachments/assets/58141804-0bc6-4b70-843d-5298ecfd4570

[SFINT-5617]: https://coveord.atlassian.net/browse/SFINT-5617?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-modal/atomic-insight-user-actions-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  public interfaceDimensions?: DOMRect;
+
+  @Prop({mutable: true}) openButton?: HTMLElement;
+  @Prop({reflect: true, mutable: true}) isOpen = false;
+  /**
+   * The ID of the user whose actions are being displayed.
+   */
+  @Prop() public userId!: string;
+  /**
+   * The date and time when the case was created..
```
Comment (SimonMilord) [inline packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-modal/atomic-insight-user-actions-modal.tsx:39]:
```suggestion
   * The date and time when the case was created.
```

**Review by SimonMilord — COMMENTED** [review]
Looks fine, Im assuming there will be tests for those components as well?

**Review by SimonMilord — COMMENTED** [review]
Will you be adding tests for the toggle and modal? Unit and e2e?

**Comment by mmitiche** [comment]
> Will you be adding tests for the toggle and modal? Unit and e2e?

Yes we have e2e for the insight user actions toggle component.

**Review by alexprudhomme — APPROVED** [review]

File: packages/atomic-react/src/components/stencil-generated/commerce/index.ts [hunk 2]
```diff
@@ -36,7 +36,9 @@ export const AtomicCommerceSortDropdown = /*@__PURE__*/createReactComponent<JSX.
 export const AtomicCommerceText = /*@__PURE__*/createReactComponent<JSX.AtomicCommerceText, HTMLAtomicCommerceTextElement>('atomic-commerce-text');
 export const AtomicCommerceTimeframeFacet = /*@__PURE__*/createReactComponent<JSX.AtomicCommerceTimeframeFacet, HTMLAtomicCommerceTimeframeFacetElement>('atomic-commerce-timeframe-facet');
 export const AtomicIcon = /*@__PURE__*/createReactComponent<JSX.AtomicIcon, HTMLAtomicIconElement>('atomic-icon');
+export const AtomicInsightUserActionsModal = /*@__PURE__*/createReactComponent<JSX.AtomicInsightUserActionsModal, HTMLAtomicInsightUserActionsModalElement>('atomic-insight-user-actions-modal');
 export const AtomicInsightUserActionsTimeline = /*@__PURE__*/createReactComponent<JSX.AtomicInsightUserActionsTimeline, HTMLAtomicInsightUserActionsTimelineElement>('atomic-insight-user-actions-timeline');
+export const AtomicInsightUserActionsToggle = /*@__PURE__*/createReactComponent<JSX.AtomicInsightUserActionsToggle, HTMLAtomicInsightUserActionsToggleElement>('atomic-insight-user-actions-toggle');
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/atomic-react/src/components/stencil-generated/commerce/index.ts:41]: The reason why I didn't mark them as internal, is because when I did, the public properties of these components were not made interactive in Storybook so I was not able to test these components:

for example, with `@internal` that's how storybook renders:
Screenshot 2024-09-12 at 11 31 51 AM

no inputs to introduce the `ticketCreationDateTime` or the `userId` parameters.

[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/atomic-react/src/components/stencil-generated/commerce/index.ts:41]: I tried tagging the component with @insight, but they still appear in generated files :/ any idea how to surpass this?

File: packages/atomic/src/components/insight/atomic-insight-history-toggle/atomic-insight-history-toggle.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ export class AtomicInsightHistoryToggle {
   @Prop({mutable: true}) public clickCallback: () => void = () => {};
 
   @Prop({mutable: true}) public tooltip = '';
+  @Prop({mutable: true}) public buttonRef?: (el?: HTMLButtonElement) => void;
```
Comment (erocheleau) [inline packages/atomic/src/components/insight/atomic-insight-history-toggle/atomic-insight-history-toggle.tsx:17]: Why is this needed?

**Review by erocheleau — APPROVED** [review]
Couple of questions, but obviously it works.

Just wondering if we should create a "preview" button instead that does nothing, and a "real" button that actually works?

File: packages/atomic/storybookUtils/insight/insight-interface-wrapper.tsx [hunk 4]
```diff
@@ -13,6 +13,16 @@ export const wrapInInsightInterface = (
...
+    <style data-styles>
+      atomic-insight-interface:not([widget='false']),
+      atomic-insight-layout:not([widget='false']) {
+        width: 500px;
+        height: 1000px;
+        margin-left: auto;
+        margin-right: auto;
+        box-shadow: 0px 3px 24px 0px #0000001a;
+      }
+    </style>
```
Comment (SimonMilord) [inline packages/atomic/storybookUtils/insight/insight-interface-wrapper.tsx:25]: why is this needed on the insight interface wrapper? For the modal?

File: packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/e2e/atomic-insight-user-actions-toggle.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    });
+  });
+
+  test('should display the user actions toggle button', async ({
+    userActionsToggle,
+  }) => {
+    await expect(userActionsToggle.atomicInsightHistoryToggle).toBeVisible();
+  });
+
+  test.describe('when clicking the user actions toggle', () => {
```
Comment (SimonMilord) [inline packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/e2e/atomic-insight-user-actions-toggle.e2e.ts:22]: we should probably add a test or a check to make sure the header is properly displayed as well on top of the timeline.

**Review by SimonMilord — APPROVED** [review]
Nice work! Love the animation!

**Comment by pathiery** [comment]
@erocheleau @lbergeron should we transfer ownership of these files to sfint ? Seeing as you are the ones working on it and we don't have much knowledge about them anymore... (I can review this now but I don't have a lot of knowledge about it)

File: packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/atomic-insight-user-actions-toggle.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  /**
+   * The date and time when the case was created. For example "2024-01-01T00:00:00Z"
+   */
+  @Prop() public ticketCreationDateTime!: string;
+
+  private buttonRef?: HTMLButtonElement;
+  private modalRef?: HTMLAtomicInsightUserActionsModalElement;
+
+  private enableModal() {
+    this.modalRef && (this.modalRef.isOpen = true);
```
Comment (pathiery) [inline packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/atomic-insight-user-actions-toggle.tsx:31]: I would probably use an `if` here just for the sake of clarity

**Review by pathiery — APPROVED** [review]
Other comments still stand, but I'll approve to allow you to work faster

File: packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/e2e/atomic-insight-user-actions-toggle.e2e.ts (hunk 5 above)
Comment (mmitiche) [reply packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-toggle/e2e/atomic-insight-user-actions-toggle.e2e.ts:22]: the Atomic Modal is responsible of displaying the header andI believe we have other tests specific to the Atomic Modal.

File: packages/atomic/storybookUtils/insight/insight-interface-wrapper.tsx (hunk 4 above)
Comment (mmitiche) [reply packages/atomic/storybookUtils/insight/insight-interface-wrapper.tsx:25]: Cause we wanna make the test like they are really in the insight panel so this is giving it the shape of the insight panel.

File: packages/atomic-react/src/components/stencil-generated/commerce/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/atomic-react/src/components/stencil-generated/commerce/index.ts]: solved!

File: packages/atomic/src/components/insight/atomic-insight-history-toggle/atomic-insight-history-toggle.tsx (hunk 3 above)
Comment (mmitiche) [reply packages/atomic/src/components/insight/atomic-insight-history-toggle/atomic-insight-history-toggle.tsx]: no longer needed👍
